src/ott/solvers/quadratic/histogram_transport.py

```diff
@@ -0,0 +1,243 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import (
+    Any,
+    Callable,
+    NamedTuple,
+    Optional,
+    Tuple,
+    Union,
+)
+
+import jax
+import jax.experimental
+import jax.numpy as jnp
+import numpy as np
+
+from ott import utils
+from ott.geometry import geometry
+from ott.problems.linear import linear_problem
+from ott.problems.quadratic import quadratic_problem
+from ott.solvers.linear import sinkhorn, sinkhorn_lr
+from ott.tools import soft_sort
+
+__all__ = ["HistogramTransport", "HTState"]
+
+LinearOutput = Union[sinkhorn.SinkhornOutput, sinkhorn_lr.LRSinkhornOutput]
+
+ProgressCallbackFn_t = Callable[
+    [Tuple[np.ndarray, np.ndarray, np.ndarray, "HTState"]], None]
+
+
+class HTOutput(NamedTuple):
```

Comment:
I don't think it's needed to create a new output class that duplicates a lot of  `GWSinkhornOutput`. We could reuse it.

Reply:
Good point. I have removed these classes.